source/libip6string/stoip6.c

```diff
@@ -20,15 +20,17 @@
 #include "ip6string.h"
 
 static uint16_t hex(const char *p);
+static bool is_hex(const char c);
```

Comment:
Not a fan of meaningless `const` in declaration prototypes - you're passing by value. You're free to retain it in the definition below, I guess, but there's no `const size_t` or `void * const dest` going on, so I think this is just randomly picked up from the line above where the `const` means something different. This is actually analogous to `const char * const p`.

(Argument variables can be declared const or not inside a function, indicating the constness of the copy. For the parameters, const is meaningless, and doesn't affect type-checking of definition versus declaration.).